### PR TITLE
Google mock

### PR DESCRIPTION
**Pull request by TJW2**

### CODE REVIEW

File: WordTreeInputTests.cpp [hunk 1]
```diff
@@ -13,6 +14,20 @@ TEST(WordTreeInput, RealSingleInput)
   EXPECT_EQ(count_of_most_common_word_from_file(file_path), 1);
 }
 
+TEST(WordTreeInput, MockSingleInput)
+{
+  std::string file_path = "test-files/test-0.txt";
```
Comment (AlexHockey) [inline WordTreeInputTests.cpp]: It would be good to give this file a more descriptive name, like `test-single-word.txt` or something.

File: WordTreeInputTests.cpp [hunk 2]
```diff
@@ -13,6 +14,20 @@ TEST(WordTreeInput, RealSingleInput)
...
+  std::pair<std::string, int> mock_pair("M",2);
```
Comment (AlexHockey) [inline WordTreeInputTests.cpp]: This pair isn't really a mock. The point of a mock is that its programmable - the test framework can control how it behaves. This is a real `std::pair` and the test doesn't get to control how the pair responds to its methods being called. 

TL;DR I would rename `mock_pair` to `iteration_result` or something.

File: WordTreeInputTests.cpp [hunk 3]
```diff
@@ -13,6 +14,20 @@ TEST(WordTreeInput, RealSingleInput)
...
+  MockWordTree mock_tree;
+  EXPECT_CALL(mock_tree, add_word("A")).Times(1);
+  add_to_tree_from_file(mock_tree, file_path);
```
Comment (AlexHockey) [inline WordTreeInputTests.cpp]: A blank line after this call would help help readability. You're setting up some expectations then calling a method that will use these expectations, then doing the same again. You want to keep the expectations visually close to the method that uses them, but distinct from other methods.

**Comment by AlexHockey** [comment]
I've reviewed this pull request. The code looks good. I've made a few stylistic comments inline. Please could you:

* Apply these markups
* Push you're changes to the `google-mock` branch. 
* Check that you can see them reflected in the PR. 
* Merge the PR and delete your branch through the github GUI.
